examples/src/main/java/com/adobe/platform/ecosystem/examples/data/write/field/converter/parquet/catalog/CatalogSchemaParquetFieldConverter.java

```diff
@@ -0,0 +1,123 @@
+ * is strictly forbidden unless prior written permission is obtained
+ * from Adobe Systems Incorporated.
+ *
+ *
+ */
+package com.adobe.platform.ecosystem.examples.data.write.field.converter.parquet.catalog;
+
+import com.adobe.platform.ecosystem.examples.catalog.model.DataType;
+import com.adobe.platform.ecosystem.examples.catalog.model.SchemaField;
+import com.adobe.platform.ecosystem.examples.constants.SDKConstants;
+import com.adobe.platform.ecosystem.examples.data.functions.FieldConverterFunction;
+import com.adobe.platform.ecosystem.examples.data.write.field.converter.parquet.ParquetFieldConverter;
+import com.adobe.platform.ecosystem.examples.parquet.model.ParquetIODataType;
+import com.adobe.platform.ecosystem.examples.parquet.model.ParquetIOField;
+import com.adobe.platform.ecosystem.examples.parquet.model.ParquetIORepetitionType;
+import com.adobe.platform.ecosystem.examples.util.ConnectorSDKException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Concrete implementation to convert
+ * SchemaFields to ParquetIOFields.
+ */
+public class CatalogSchemaParquetFieldConverter implements ParquetFieldConverter<List<SchemaField>> {
+
+    @Override
+    public List<ParquetIOField> convert(List<SchemaField> data) throws ConnectorSDKException {
+        if (data == null) {
+            throw new ConnectorSDKException("Schema Fields are null. Please provide schema.");
+        }
+        List<ParquetIOField> fields = new ArrayList<>();
+        for (SchemaField schemaField : data) {
+            fields.add(getParquetIOField(schemaField));
+        }
+        return fields;
+    }
+
+    private ParquetIOField getParquetIOField(SchemaField schemaField) throws ConnectorSDKException {
+        switch (schemaField.getType()) {
+            case Field_MapType:
+                return createMapField(schemaField);
+            case Field_ArrayType:
+                return createArrayField(schemaField);
+            case Field_ObjectType:
+                return createStructField(schemaField);
+            default:
+                // Primitive type.
+                return new ParquetIOField(
+                    schemaField.getName(),
+                    FieldConverterFunction.catalogToParquetFieldFunction.apply(schemaField.getType()),
+                    ParquetIORepetitionType.OPTIONAL,
+                    null
+                );
+        }
+    }
+
+    private ParquetIOField createStructField(SchemaField schemaField) throws ConnectorSDKException {
+        return new ParquetIOField(
+            schemaField.getName(),
+            ParquetIODataType.GROUP,
+            ParquetIORepetitionType.OPTIONAL,
+            convert(schemaField.getSubFields())
+        );
+    }
+
+    private ParquetIOField createArrayField(SchemaField schemaField) throws ConnectorSDKException {
+        // TODO: This will blow up when support for Array of Array and Array of Map support
```

Comment:
Remove this TODO as I see support for List of List/Map will be extended seamlessly.